### DISCUSSION TITLE
brew install sqitch not completed properly(Error:SIGTERM occured while./Build bundle)

### DISCUSSION THREAD

**Discussion by RutujaMGholap** [Homebrew/discussions#573] [Getting started]
rgholap@C02DRB64MD6M ~ % brew install sqitch
Updating Homebrew...
==> Installing sqitch from sqitchers/sqitch
==> Downloading https://homebrew.bintray.com/bottles/perl-5.32.0.catalina.bottle
==> Downloading from https://d29vzk4ow07wi7.cloudfront.net/bc6c97521b6edf723c8ee
######################################################################## 100.0%
==> Downloading https://github.com/miyagawa/cpanminus/archive/1.9019.tar.gz
==> Downloading from https://codeload.github.com/miyagawa/cpanminus/tar.gz/1.901
  # #=O=#   #                                                                 
==> Downloading https://www.cpan.org/authors/id/D/DW/DWHEELER/App-Sqitch-v1.1.0.
######################################################################## 100.0%
==> Installing dependencies for sqitchers/sqitch/sqitch: perl and cpanminus
==> Installing sqitchers/sqitch/sqitch dependency: perl
==> Pouring perl-5.32.0.catalina.bottle.tar.gz
==> Caveats
By default non-brewed cpan modules are installed to the Cellar. If you wish
for your modules to persist across updates we recommend using `local::lib`.

You can set that up like this:
  PERL_MM_OPT="INSTALL_BASE=$HOME/perl5" cpan local::lib
  echo 'eval "$(perl -I$HOME/perl5/lib/perl5 -Mlocal::lib=$HOME/perl5)"' >> ~/.zshrc
==> Summary
🍺  /usr/local/Cellar/perl/5.32.0: 2,458 files, 63.5MB
==> Installing sqitchers/sqitch/sqitch dependency: cpanminus
🍺  /usr/local/Cellar/cpanminus/1.9019: 4 files, 560.8KB, built in 14 seconds
==> Installing sqitchers/sqitch/sqitch
==> cpanm --local-lib instutil --notest Menlo::CLI::Compat Module::Build
==> perl Build.PL --install_base /usr/local/Cellar/sqitch/v1.1.0 --etcdir /usr/l
==> ./Build bundle
Error: SIGTERM
Please report this issue:
  https://docs.brew.sh/Troubleshooting
/usr/local/Homebrew/Library/Homebrew/utils/fork.rb:73:in `read'
/usr/local/Homebrew/Library/Homebrew/utils/fork.rb:73:in `block (3 levels) in safe_fork'
/usr/local/Homebrew/Library/Homebrew/utils.rb:405:in `ignore_interrupts'
/usr/local/Homebrew/Library/Homebrew/utils/fork.rb:63:in `block (2 levels) in safe_fork'
/usr/local/Homebrew/Library/Homebrew/utils/fork.rb:34:in `open'
/usr/local/Homebrew/Library/Homebrew/utils/fork.rb:34:in `block in safe_fork'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/tmpdir.rb:93:in `mktmpdir'
/usr/local/Homebrew/Library/Homebrew/utils/fork.rb:33:in `safe_fork'
/usr/local/Homebrew/Library/Homebrew/formula_installer.rb:912:in `build'
/usr/local/Homebrew/Library/Homebrew/formula_installer.rb:466:in `install'
/usr/local/Homebrew/Library/Homebrew/cmd/install.rb:388:in `install_formula'
/usr/local/Homebrew/Library/Homebrew/cmd/install.rb:301:in `block in install'
/usr/local/Homebrew/Library/Homebrew/cmd/install.rb:299:in `each'
/usr/local/Homebrew/Library/Homebrew/cmd/install.rb:299:in `install'
/usr/local/Homebrew/Library/Homebrew/brew.rb:122:in `<main>'
rgholap@C02DRB64MD6M ~ % Traceback (most recent call last):
	19: from /usr/local/Homebrew/Library/Homebrew/brew.rb:122:in `<main>'
	18: from /usr/local/Homebrew/Library/Homebrew/cmd/install.rb:299:in `install'
	17: from /usr/local/Homebrew/Library/Homebrew/cmd/install.rb:299:in `each'
	16: from /usr/local/Homebrew/Library/Homebrew/cmd/install.rb:301:in `block in install'
	15: from /usr/local/Homebrew/Library/Homebrew/cmd/install.rb:388:in `install_formula'
	14: from /usr/local/Homebrew/Library/Homebrew/formula_installer.rb:466:in `install'
	13: from /usr/local/Homebrew/Library/Homebrew/formula_installer.rb:912:in `build'
	12: from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:33:in `safe_fork'
	11: from /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/tmpdir.rb:93:in `mktmpdir'
	10: from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:34:in `block in safe_fork'
	 9: from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:34:in `open'
	 8: from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:37:in `block (2 levels) in safe_fork'
	 7: from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:37:in `fork'
	 6: from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:42:in `block (3 levels) in safe_fork'
	 5: from /usr/local/Homebrew/Library/Homebrew/formula_installer.rb:924:in `block in build'
	 4: from /usr/local/Homebrew/Library/Homebrew/sandbox.rb:99:in `exec'
	 3: from /usr/local/Homebrew/Library/Homebrew/utils.rb:302:in `safe_system'
	 2: from /usr/local/Homebrew/Library/Homebrew/utils.rb:47:in `system'
	 1: from /usr/local/Homebrew/Library/Homebrew/utils.rb:38:in `_system'
/usr/local/Homebrew/Library/Homebrew/utils.rb:38:in `wait': SIGTERM (SignalException)
	16: from /usr/local/Homebrew/Library/Homebrew/brew.rb:122:in `<main>'
	15: from /usr/local/Homebrew/Library/Homebrew/cmd/install.rb:299:in `install'
	14: from /usr/local/Homebrew/Library/Homebrew/cmd/install.rb:299:in `each'
	13: from /usr/local/Homebrew/Library/Homebrew/cmd/install.rb:301:in `block in install'
	12: from /usr/local/Homebrew/Library/Homebrew/cmd/install.rb:388:in `install_formula'
	11: from /usr/local/Homebrew/Library/Homebrew/formula_installer.rb:466:in `install'
	10: from /usr/local/Homebrew/Library/Homebrew/formula_installer.rb:912:in `build'
	 9: from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:33:in `safe_fork'
	 8: from /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/tmpdir.rb:93:in `mktmpdir'
	 7: from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:34:in `block in safe_fork'
	 6: from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:34:in `open'
	 5: from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:37:in `block (2 levels) in safe_fork'
	 4: from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:37:in `fork'
	 3: from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:38:in `block (3 levels) in safe_fork'
	 2: from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:55:in `rescue in block (3 levels) in safe_fork'
	 1: from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:55:in `puts'
/usr/local/Homebrew/Library/Homebrew/utils/fork.rb:55:in `write': Broken pipe (Errno::EPIPE)
Traceback (most recent call last):
	23: from /usr/local/Homebrew/Library/Homebrew/build.rb:223:in `<main>'
	22: from /usr/local/Homebrew/Library/Homebrew/build.rb:134:in `install'
	21: from /usr/local/Homebrew/Library/Homebrew/utils.rb:531:in `with_env'
	20: from /usr/local/Homebrew/Library/Homebrew/build.rb:139:in `block in install'
	19: from /usr/local/Homebrew/Library/Homebrew/formula.rb:1215:in `brew'
	18: from /usr/local/Homebrew/Library/Homebrew/formula.rb:2236:in `stage'
	17: from /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/forwardable.rb:230:in `stage'
	16: from /usr/local/Homebrew/Library/Homebrew/resource.rb:91:in `stage'
	15: from /usr/local/Homebrew/Library/Homebrew/resource.rb:116:in `unpack'
	14: from /usr/local/Homebrew/Library/Homebrew/resource.rb:197:in `mktemp'
	13: from /usr/local/Homebrew/Library/Homebrew/mktemp.rb:63:in `run'
	12: from /usr/local/Homebrew/Library/Homebrew/mktemp.rb:63:in `chdir'
	11: from /usr/local/Homebrew/Library/Homebrew/mktemp.rb:63:in `block in run'
	10: from /usr/local/Homebrew/Library/Homebrew/resource.rb:121:in `block in unpack'
	 9: from /usr/local/Homebrew/Library/Homebrew/formula.rb:2256:in `block in stage'
	 8: from /usr/local/Homebrew/Library/Homebrew/utils.rb:531:in `with_env'
	 7: from /usr/local/Homebrew/Library/Homebrew/formula.rb:2257:in `block (2 levels) in stage'
	 6: from /usr/local/Homebrew/Library/Homebrew/formula.rb:1222:in `block in brew'
	 5: from /usr/local/Homebrew/Library/Homebrew/build.rb:171:in `block (2 levels) in install'
	 4: from /usr/local/Homebrew/Library/Taps/sqitchers/homebrew-sqitch/Formula/sqitch.rb:105:in `install'
	 3: from /usr/local/Homebrew/Library/Homebrew/formula.rb:2043:in `system'
	 2: from /usr/local/Homebrew/Library/Homebrew/formula.rb:2043:in `open'
	 1: from /usr/local/Homebrew/Library/Homebrew/formula.rb:2084:in `block in system'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2084:in `wait': SIGTERM (SignalException)
	2: from /usr/local/Homebrew/Library/Homebrew/build.rb:211:in `<main>'
	1: from /usr/local/Homebrew/Library/Homebrew/build.rb:244:in `rescue in <main>'
/usr/local/Homebrew/Library/Homebrew/build.rb:244:in `close': Broken pipe (Errno::EPIPE)

please help me in resolving this issue

**Comment by carlocab** [comment]
What does `brew config` and `brew doctor` say?

**Reply by RutujaMGholap** [reply]
rgholap@C02DRB64MD6M ~ % brew config
HOMEBREW_VERSION: 2.7.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: bca4804a9e48de5319383d3eddadaa7f054c77da
Last commit: 4 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 6e04d4a051e15c0dbfab3ad17894908e58691008
Core tap last commit: 61 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
CPU: dodeca-core 64-bit kabylake
Clang: 12.0 build 1200
Git: 2.24.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 10.15.7-x86_64
CLT: 12.0.0.0.1.1599194153
Xcode: N/A

**Reply by RutujaMGholap** [reply]
rgholap@C02DRB64MD6M ~ % brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Preferences or run:
  softwareupdate --all --install --force

If that doesn't show you an update run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/more/.

**Reply by carlocab** [reply]
Try updating your Command Line Tools as in the instructions from `brew doctor`. Then, do `brew update` before trying `brew install` again.

**Reply by RutujaMGholap** [reply]
Okk..I'll try updating CLT .

**Reply by RutujaMGholap** [reply]
rgholap@C02DRB64MD6M ~ % brew doctor
Your system is ready to brew.
rgholap@C02DRB64MD6M ~ % brew update
Already up-to-date.
rgholap@C02DRB64MD6M ~ % brew install sqitch
Updating Homebrew...
==> Installing sqitch from sqitchers/sqitch
==> Downloading https://www.cpan.org/authors/id/D/DW/DWHEELER/App-Sqitch-v1.1.0.
Already downloaded: /Users/rgholap/Library/Caches/Homebrew/downloads/f1ebd13dbda2c5ad95c6f9f14278d7771011f3d113aaa624d6fb79e78bd7fb3c--App-Sqitch-v1.1.0.tar.gz
==> cpanm --local-lib instutil --notest Menlo::CLI::Compat Module::Build
==> perl Build.PL --install_base /usr/local/Cellar/sqitch/v1.1.0 --etcdir /usr/l
==> ./Build bundle

Last 15 lines from /Users/rgholap/Library/Logs/Homebrew/sqitch/03.Build:
OK
tar: Error opening archive: truncated gzip input
--> Working on Template
Fetching http://www.cpan.org/authors/id/A/AT/ATOOMIC/Template-Toolkit-3.009.tar.gz ... ! Failed to unpack DateTime-1.54.tar.gz: no directory
! Failed to fetch distribution DateTime-1.54
OK
tar: truncated gzip input
tar: Error exit delayed from previous errors.
! No MYMETA file is found after configure. Your toolchain is too old?
! The distribution doesn't have a proper Makefile.PL/Build.PL See /private/tmp/sqitch-20210119-11552-1v9t1yo/App-Sqitch-v1.1.0/.brew_home/.cpanm/work/1611039045.16332/build.log for details.
! Installing the dependencies failed: Module 'DateTime' is not installed, Module 'String::Formatter' is not installed, Module 'Locale::TextDomain' is not installed, Module 'Locale::Messages' is not installed, Module 'List::MoreUtils' is not installed, Module 'Sub::Exporter::Util' is not installed, Module 'Clone' is not installed, Module 'IO::Pager' is not installed, Module 'Pod::Find' is not installed, Module 'DBI' is not installed, Module 'Sub::Exporter' is not installed, Module 'Throwable' is not installed, Module 'StackTrace::Auto' is not installed
! Bailing out the installation for App-Sqitch-v1.1.0.
Error installing modules: 
Configuring Template-Toolkit-3.009 ... N/A
63 distributions installed

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
  https://github.com/sqitchers/homebrew-sqitch/issues

These open issues may also help:
After brew install sqitch --with-postgres-support, error deploying https://github.com/sqitchers/homebrew-sqitch/issues/46
rgholap@C02DRB64MD6M ~ % 
rgholap@C02DRB64MD6M ~ % sqitch 
zsh: command not found: sqitch

**Reply by carlocab** [reply]
This is a problem with the formula, and Homebrew has no control over that because it's in a third-party tap. That's why the output reads:
```
If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
https://github.com/sqitchers/homebrew-sqitch/issues
```
Please report this at https://github.com/sqitchers/homebrew-sqitch/issues. You should check that there are no prior issues reporting the same bug first, however.
